### PR TITLE
Assignment-7 done

### PR DESCRIPTION
**Pull request by nikhil1-successive**

### CODE REVIEW

File: Assignment-7/ques4.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      fetchAlbums()
+        .then((data) => {
+          apiCallCnt++;
+          lastApiCallTimestamp = currentTimestamp;
+          resolve(data);
+        })
+        .catch((err) => {
+          reject(err);
+        });
+    } else if (currentTimestamp - lastApiCallTimestamp < intervalInMillisec) {
```
Comment (shahbaz-successive) [inline Assignment-7/ques4.js:22]: else if condition is unnecessary so remove and reject in else

File: Assignment-7/ques4.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  const apiPromise = callApi();
+
+  apiPromise
+    .then((data) => {
+      console.log(data);
+      setTimeout(fetchApis, 1000);
+    })
+    .catch((err) => {
+      console.error("Error:", err, "Retry after 10 seconds.");
+      setTimeout(fetchApis, 1000);
```
Comment (shahbaz-successive) [inline Assignment-7/ques4.js:72]: Don't call API again when error encountered

File: Assignment-7/ques4.js (hunk 1 above)
Comment (nikhil1-successive) [reply Assignment-7/ques4.js:22]: done

File: Assignment-7/ques4.js (hunk 2 above)
Comment (nikhil1-successive) [reply Assignment-7/ques4.js:72]: done

**Review by shahbaz-successive — APPROVED** [review]
👍

**Pull request merged**